### DISCUSSION TITLE
Localization - Multilingual

### DISCUSSION THREAD

**Discussion by chrisblockchainthai** [lunarphp/lunar#354] [Feature Requests]
What's the point of having language in settings when 98% of the site isn't even translatable? It's a lot of work to get all the text into the language files but that's what would actually separate this project from every other! It's not just the language however it's the localization too, the way numbers and dates are shown need also to be localized otherwise out of the box this project is only useful in the USA.

* seo friendly urls /locahost/en-en/  localhost/en-th/
* localiztion of dates and currency
* LTR

**Comment by glennjacobs** [comment]
Hey there. 

I'm not sure if you are referring to the demo store here or the admin hub or what exactly, so I'll try my best. 

The demo store is just that, a code demo of how you could use GetCandy. GetCandy is headless e-commerce, so the demo store isn't really a product as such, just there as an example. 

The admin hub uses language files and hopefully we'll have contributions in the future to translate it into different languages by the community. 

Languages are there primarily for three reasons. One, so you can create translatable attributes on products, collections, etc. Two, so that you can make translatable slugs for products etc and three, so you have a reference to implement them on your custom frontend.

**Comment by chrisblockchainthai** [comment]
ok so i get your point the demo is just to show y9ou how it could work. But  these other vendor packages like livewire Converter.php you got  codes like this:
 
 throw new Exception("Measurement [{$measurement}] was not found.");

i realise it's not your package but inherently it is a problem cause your admin area will still throw up english when i don't want it to.

What's your suggestion?

**Comment by glennjacobs** [comment]
PHP exceptions are an interesting one. I think these edge-cases would have to be dealt with individually by raising a GitHub issue on this repo and seeing what solutions we have. 

If you have an urgent need for something, then you are welcome to suggest a fix for the issue via a pull request.
